sql/core/src/main/scala/org/apache/spark/sql/streaming/progress.scala

```diff
@@ -163,7 +163,8 @@ class SourceProgress protected[sql](
   val endOffset: String,
   val numInputRows: Long,
   val inputRowsPerSecond: Double,
-  val processedRowsPerSecond: Double) extends Serializable {
+  val processedRowsPerSecond: Double,
+  val customMetrics: Option[JValue] = None) extends Serializable {
```

Comment:
Default value does not work in Java API and probably MiMa complains about this. I think another constructor should better be made instead of default value to work around this.